File: DevSupportAndroid/src/main/java/com/sup/dev/android/tools/ToolsToast.java
package com.sup.dev.android.tools;

import android.support.annotation.StringRes;
import android.support.design.widget.Snackbar;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.sup.dev.android.app.SupAndroid;
import com.sup.dev.java.tools.ToolsThreads;

public class ToolsToast {

    public static void show(@StringRes int textRes) {
        show(SupAndroid.appContext.getString(textRes));
    }

    public static void show(@StringRes int textRes, Object... args) {
        show(ToolsResources.getString(textRes, args));
    }

    public static void show(String text) {
        ToolsThreads.main(() -> showNow(text));
    }

    private static void showNow(String text) {
        if (text == null || text.isEmpty()) return;
        Toast.makeText(SupAndroid.appContext, text, Toast.LENGTH_SHORT).show();
    }

    public static void showSnack(@StringRes int textRes) {
        showSnack(SupAndroid.appContext.getString(textRes));
    }

    public static void showSnack(String text) {
        ToolsThreads.main(() -> showSnackNow(SupAndroid.activity.getViewContainer(), text));
    }

    public static void showSnack(View v, @StringRes int textRes) {
        showSnack(v, SupAndroid.appContext.getString(textRes));
    }

    public static void showSnack(View v, String text) {
        ToolsThreads.main(() -> showSnackNow(v, text));
    }

    private static void showSnackNow(View v, String text) {
        if (text == null || text.isEmpty()) return;
        Snackbar snack = Snackbar.make(v, text, Snackbar.LENGTH_SHORT);
        View snackbarView = snack.getView();
        TextView textView = snackbarView.findViewById(android.support.design.R.id.snackbar_text);
        textView.setMaxLines(5);
        snack.show();
    }

    public static void showSnackIfLocked(View v, @StringRes int textRes) {
        showSnackIfLocked(v, ToolsResources.getString(textRes));
    }

    public static void showSnackIfLocked(View v, String text) {
        ToolsThreads.main(() -> {
            if (ToolsAndroid.isScreenKeyLocked()) showSnackNow(v, text);
            else showNow(text);
        });
    }


}

File: DevSupportAndroid/src/main/java/com/sup/dev/android/tools/ToolsToast.kt
package com.sup.dev.android.tools

import android.support.annotation.StringRes
import android.support.design.widget.Snackbar
import android.view.View
import android.widget.TextView
import android.widget.Toast
import com.sup.dev.android.app.SupAndroid
import com.sup.dev.java.tools.ToolsThreads


object ToolsToast {

    fun showToast(@StringRes textRes: Int) {
        showToast(SupAndroid.appContext!!.getString(textRes))
    }

    fun showToast(@StringRes textRes: Int, vararg args: Any) {
        showToast(ToolsResources.getString(textRes, *args))
    }

    fun showToast(text: String?) {
        ToolsThreads.main { showToastNow(text) }
    }

    private fun showToastNow(text: String?) {
        if (text == null || text.isEmpty()) return
        Toast.makeText(SupAndroid.appContext!!, text, Toast.LENGTH_SHORT).show()
    }

    fun show(@StringRes textRes: Int) {
        show(SupAndroid.appContext!!.getString(textRes))
    }

    fun show(text: String) {
        ToolsThreads.main { showNow(SupAndroid.activity!!.getViewContainer()!!, text) }
    }

    fun show(v: View, @StringRes textRes: Int) {
        show(v, SupAndroid.appContext!!.getString(textRes))
    }

    fun show(v: View, text: String) {
        ToolsThreads.main { showNow(v, text) }
    }

    private fun showNow(v: View, text: String?) {
        if (text == null || text.isEmpty()) return
        val snack = Snackbar.make(v, text, Snackbar.LENGTH_SHORT)
        val snackbarView = snack.view
        val textView = snackbarView.findViewById<TextView>(android.support.design.R.id.snackbar_text)
        textView.maxLines = 5
        snack.show()
    }

    fun showIfLocked(v: View, @StringRes textRes: Int) {
        showIfLocked(v, ToolsResources.getString(textRes))
    }

    fun showIfLocked(v: View, text: String?) {
        ToolsThreads.main {
            if (ToolsAndroid.isScreenKeyLocked())
                showNow(v, text)
            else
                showToastNow(text)
        }
    }


}
